Client/Ecosia/UI/WhatsNew/DataProvider/WhatsNewLocalDataProvider.swift

```diff
@@ -22,39 +22,31 @@ final class WhatsNewLocalDataProvider: WhatsNewDataProvider {
     /// A computed property to determine whether the "What's New" page should be displayed.
     /// - Returns: `true` if the What's New page should be shown; otherwise, `false`.
     var shouldShowWhatsNewPage: Bool {
-        
-        // Check if we are in the upgrade scenario
         guard EcosiaInstallType.get() == .upgrade else {
-            markAllPreviousVersionsAsSeen()
+            markPreviousVersionsAsSeen()
             return false
         }
         
-        // Get a list of version strings from the data provider.
-        let dataProviderVersionsString = getVersionRange().map { $0.description }
+        // Are there items to be shown in the range?
+        guard let items = try? getData(), !items.isEmpty else { return false }
```

Comment:
not exactly part of this PR but `getData()` is not really accurate on what the method does imo. Could we rename it to `getWhatsNewItems()`?

Reply:
Makes sense! We'll even update it to `getWhatsNewItemsInRange` to make it explicit that we do check the range ✅